File: LLM-Translations/GPT-4/avatar/Python/Java/atcoder_ABC162_D.java
import java.util.Scanner;

public class atcoder_ABC162_D {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        String S = scanner.next();
        calculate(N, S);
    }

    public static void calculate(int n, String s) {
        char[] arr = s.toCharArray();
        int rNum = 0, gNum = 0, bNum = 0;
        for (char c : arr) {
            if (c == 'R') rNum++;
            else if (c == 'G') gNum++;
            else if (c == 'B') bNum++;
        }
        int sum = 0;
        for (int step = 1; step <= Math.ceil(n / 2.0); step++) {
            for (int i = 0; i < n - 2 * step; i++) {
                String sCombination = "" + arr[i] + arr[i + step] + arr[i + step * 2];
                if ("RGB".equals(sCombination) || "RBG".equals(sCombination) || "BGR".equals(sCombination) || "BRG".equals(sCombination) || "GBR".equals(sCombination) || "GRB".equals(sCombination)) {
                    sum++;
                }
            }
        }
        System.out.println(rNum * gNum * bNum - sum);
    }
}
// End of Code

File: LLM-Translations/GPT-4/avatar/Python/C/atcoder_ABC162_D.c
#include <stdio.h>
#include <string.h>
#include <math.h>

void calculate(int n, char* s) {
    int rNum = 0, gNum = 0, bNum = 0, sum = 0, i, step;
    for(i = 0; i < n; i++) {
        if(s[i] == 'R') rNum++;
        else if(s[i] == 'G') gNum++;
        else if(s[i] == 'B') bNum++;
    }
    for(step = 1; step <= ceil(n / 2.0); step++) {
        for(i = 0; i < n - 2 * step; i++) {
            if((s[i] == 'R' && s[i + step] == 'G' && s[i + step * 2] == 'B') ||
               (s[i] == 'R' && s[i + step] == 'B' && s[i + step * 2] == 'G') ||
               (s[i] == 'B' && s[i + step] == 'R' && s[i + step * 2] == 'G') ||
               (s[i] == 'B' && s[i + step] == 'G' && s[i + step * 2] == 'R') ||
               (s[i] == 'G' && s[i + step] == 'B' && s[i + step * 2] == 'R') ||
               (s[i] == 'G' && s[i + step] == 'R' && s[i + step * 2] == 'B')) {
                sum++;
            }
        }
    }
    printf("%d\n", rNum * gNum * bNum - sum);
}

int main() {
    int N;
    char S[1000];
    scanf("%d", &N);
    scanf("%s", S);
    calculate(N, S);
    return 0;
}
// End of Code
